http/messages.go

```diff
@@ -0,0 +1,149 @@
+package http
+
+import (
+	"sort"
+
+	"github.com/kkrull/gohttp/msg/clienterror"
+	"github.com/kkrull/gohttp/msg/servererror"
+)
+
+func NewGetMessage(path string) RequestMessage {
+	return &requestMessage{
+		method: "GET",
+		target: path,
+		path:   path,
+	}
+}
+
+func NewHeadMessage(path string) RequestMessage {
+	return &requestMessage{
+		method: "HEAD",
+		target: path,
+		path:   path,
+	}
+}
+
+func NewOptionsMessage(target string) RequestMessage {
```

Comment:
You're right - there are a few places I need to update for the other methods, since just about everything operates on the path portion of the target instead of the whole target.

This case does happen to be an edge case, where `OPTIONS * HTTP/1.1` (note the lack of `/` in the target) is a valid message: https://tools.ietf.org/html/rfc7230#section-5.3.4.  That's why there's a different name for this one, although maybe some better naming could further clarify the intent.